eslint.config.mjs

```diff
@@ -0,0 +1,58 @@
+import globals from "globals";
+import tseslint from "typescript-eslint";
+import pluginReact from "eslint-plugin-react";
+import eslint from "@eslint/js";
+import eslintPluginPrettierRecommended from "eslint-plugin-prettier/recommended";
+import { fileURLToPath } from "node:url";
+import { includeIgnoreFile } from "@eslint/compat";
+import pluginRouter from "@tanstack/eslint-plugin-router";
+
+export default tseslint.config(
+  {
+    ignores: ["**/*.mjs", "**/*.cjs", "**/*.js"],
+  },
+  includeIgnoreFile(fileURLToPath(new URL(".gitignore", import.meta.url))),
+  ...pluginRouter.configs["flat/recommended"],
+  pluginReact.configs.flat.recommended,
+  pluginReact.configs.flat["jsx-runtime"],
+  eslint.configs.recommended,
+  ...tseslint.configs.strictTypeChecked,
+  ...tseslint.configs.stylisticTypeChecked,
+  eslintPluginPrettierRecommended,
+  {
+    languageOptions: {
+      ...pluginReact.configs.flat.recommended.languageOptions,
+      globals: {
+        ...globals.node,
+        ...globals.jest,
+        ...globals.browser,
+      },
+      ecmaVersion: 5,
+      sourceType: "module",
+      parserOptions: {
+        projectService: true,
+        tsconfigRootDir: import.meta.dirname,
+      },
+    },
+    settings: {
+      react: {
+        version: "detect",
```

Comment:
Optional: this complains about not having a version of react present in this package.json.

It might be worth adding React here just to ensure this is using the right version of the react linting rules? Then syncpack can ensure that the react versions are consistent.

Reply:
Thanks, I've added it to the root-level package.json